src/emulator/storage/index.ts

```diff
@@ -113,6 +113,16 @@ export class StorageEmulator implements EmulatorInstance {
     return this._app!;
   }
 
+  async replaceRules(rules: SourceFile | RulesConfig[]): Promise<StorageRulesIssues> {
+    await this._rulesManager.stop();
+    this.createRulesManager(rules);
+    return this._rulesManager.start();
+  }
+
+  private createRulesManager(rules: SourceFile | RulesConfig[]): void {
+    this._rulesManager = createStorageRulesManager(rules, this._rulesRuntime);
```

Comment:
small nit: Would prefer we minimize side effects and remove them where possible. See go/java-practices/methods#side-effects. 

e.g. compare:
```
this._rulesManager = createRulesManager(...);
// vs 
createRulesManager(...);
```

Reply:
While this isn't necessarily too unexpected from a method named `create...()`, it would still be nice to make mutation explicit. Another resource: go/java-practices/immutability

Reply:
Done.